bsd/enclosure.pyx

```diff
@@ -109,10 +112,12 @@ cdef class Enclosure(object):
                 objd.elm_desc_len = UINT16_MAX
                 objd.elm_desc_str = <char*>calloc(UINT16_MAX, sizeof(char))
                 if not objd.elm_desc_str:
+                    self.clean_up(num_elms, elm_info, objp)
                     raise MemoryError('calloc objd.elm_desc_str failed')
                 else:
                     res = ioctl(self.enc_fd, ses.ENCIOC_GETELMDESC, <ses.caddr_t>&objd)
                     if res < 0:
+                        self.clean_up(num_elms, elm_info, objp)
                         raise RuntimeError('ioctl failed to get element description')
                     else:
                         elm_info[i].elm_desc_str = objd.elm_desc_str
```

Comment:
For this to work correctly you have to copy these pointers before the call to clean_up. Even when the ioctl fails we have to free the allocation, and the easiest way is for the pointer to be in elm_info. Likewise below on line 136.

Reply:
missed the local commit since I'm copying the file to real hardware, compiling, then testing blah blah. Anyways, fixed it.